pyanaconda/payload/__init__.py

```diff
@@ -137,6 +139,89 @@ class DependencyError(PayloadError):
 class PayloadInstallError(PayloadError):
     pass
 
+PayloadRequirement = namedtuple('PayloadRequirement', ['id', 'reasons', 'strong'])
+
+class PayloadRequirements(object):
+    """A container for payload requirements imposed by installed functionality
+
+    Stores names of packages and groups required by used installer features,
+    together with descriptions of reasons why the object is required and if the
+    requirement is strong. Not satisfying strong requirement would be fatal for
+    installation.
+    """
+
+    def __init__(self):
+        self._reqs = {}
+        for req_type in PayloadRequirementType:
+            self._reqs[req_type] = OrderedDict()
+
+    def add_packages(self, package_names, reason, strong=True):
+        """Add packages required for the reason.
+
+        If a package is already required, the new reason will be
+        added and the strength of the requirement will be updated.
+
+        :param package_names: names of packages to be added
+        :type package_names: list of str
+        :param reason: description of reason for adding the packages
+        :type reason: str
+        :param strong: is the requirement strong (ie is not satisfying it fatal?)
+        :type strong: bool
+        """
+        self._add(PayloadRequirementType.package, package_names, reason, strong)
+
+    def add_groups(self, group_ids, reason, strong=True):
+        """Add groups required for the reason.
+
+        If a group is already required, the new reason will be
+        added and the strength of the requirement will be updated.
+
+        :param group_ids: ids of groups to be added
+        :type group_ids: list of str
+        :param reason: descripiton of reason for adding the groups
+        :type reason: str
+        :param strong: is the requirement strong
```

Comment:
Could you please add the `` (ie is not satisfying it fatal?)`` message here too.